src/registries/npm-registry.js

```diff
@@ -290,6 +269,11 @@ export default class NpmRegistry extends Registry {
     }
 
     const baseRegistry = this.getRegistry(packageIdent);
+
+    if (!baseRegistry) {
+      return '';
+    }
```

Comment:
Now that we more strictly check what registry the request is for (e.g. don't conclude a request to `https://github.com` is to known registry), we can conclude that non registry requests don't need auth. This is basically the replacement for the `isRequestToRegistry`.